arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,350 @@
+					linux,default-trigger = "a5_red";
+				};
+				led-act {
+					status = "disabled";
+				};
+			};
+
+			gpio-keys {
+				compatible = "gpio-keys";
+				pinctrl-names = "default";
+				pinctrl-0 = <&button_pins>;
+				user-event {
+					label = "User event key";
+					gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+					linux,code = <KEY_UNKNOWN>;
+				};
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pinctrl-names = "default";
+
+			spi5_cs_pins {
+				/* tpm ksz*/
+				brcm,pins     = <17 18>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			spi6_cs_pins {
+				/* aout ain mbus */
+				brcm,pins     = <6 10 27>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			relais_pins: relais_pins {
+				brcm,pins     = <28>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			button_pins: button_pins {
+				brcm,pins     = <23>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			tpm_pins: tpm_pins {
+				/* irq */
+				brcm,pins     = <22>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			rs485_0_pins: rs485_0_pins {
+				/* tx rx rts */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+			rs485_1_pins: rs485_1_pins {
+				/* tx rx rts */
+				brcm,pins     = <8 9 11>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_DOWN>;
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&i2c6>;
+		__overlay__ {
+			// TODO: ANYTHING ELSE REQUIRED HERE?
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c6_gpio0>;
+			status = "okay";
+		};
+	};
+
+	fragment@3 {
+		target = <&i2c1>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c1_gpio2>;
+			status = "okay";
+
+			expander: gpio@20 {
+				compatible = "ti,tca6416";
+				// TODO: which reg to use here?
+				reg = <0x20>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				status = "okay";
+			};
+
+			rtc@51 {
+				compatible = "nxp,pcf2129";
+				// TODO: which reg to use here?
+				reg = <0x51>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@4 {
+		target = <&spi5>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			cs-gpios = <&gpio 17 GPIO_ACTIVE_LOW>,
+				   <&gpio 18 GPIO_ACTIVE_LOW>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&spi5_pins &spi5_cs_pins>;
+			status = "okay";
+
+			tpm: tpm@0 {
+				compatible = "infineon,slb9670";
+				pinctrl-names = "default";
+				pinctrl-0 = <&tpm_pins>;
+				reg = <0>;
+				spi-max-frequency = <1000000>;	/* <= 43 MHz */
+				interrupt-parent = <&gpio>;
+				#interrupt-cells = <2>;
+				interrupts = <22 IRQ_TYPE_LEVEL_LOW>;
+				status = "okay";
+			};
+			ksz@1 {
```

Comment:
`spi-max-frequency` is missing

Reply:
The examples in <a href="https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/net/micrel-ks8995.txt">Documentation/devicetree/bindings/net/micrel-ks8995.txt</a> and <a href="https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/net/dsa/microchip,ksz.yaml">Documentation/devicetree/bindings/net/dsa/microchip,ksz.yaml</a> suggest `switch@1`?

Reply:
May need to revisit whether the switch and its ports shall be represented in the DeviceTree. A TODO may make sense as a reminder.